docs/book/middleware.md

```diff
@@ -0,0 +1,53 @@
+# PSR-7 Middleware
+For every bus system a middleware exists and one Middleware to rule them all. If you use JSON or XML in the request body
+for your message data, you have to convert this data to an array before you can call the middleware.
+
+> Note: The middleware uses an array for the message data
+
+## CommandMiddleware
+The `CommandMiddleware` dispatches the message data to the command bus system. This middleware needs an request attribute 
+(`$request->getAttribute(\Prooph\Psr7Middleware\CommandMiddleware::NAME_ATTRIBUTE)`) called `prooph_command_name`. 
+This name is used for the `\Prooph\Common\Messaging\MessageFactory` to create the `\Prooph\Common\Messaging\Message` 
+object. The data for the command is extracted from the body of the request (`$request->getParsedBody()`) and must be an 
+array.
+
+## QueryMiddleware
+The `QueryMiddleware` dispatches the message data to the query bus system. This middleware needs an request attribute 
+(`$request->getAttribute(\Prooph\Psr7Middleware\QueryMiddleware::NAME_ATTRIBUTE)`) called `prooph_query_name`. 
+This name is used for the `\Prooph\Common\Messaging\MessageFactory` to create the `\Prooph\Common\Messaging\Message` 
+object. The data for the query is extracted from the body of the request (`$request->getParsedBody()`) and must be an 
```

Comment:
@prolic as far as I remember your ZF2-Query-API uses GET (query) params to populate query payload, right? I think we should do the same in the Query-Middleware.
Thoughts? @sandrokeil @prolic